CMakeLists.txt

```diff
@@ -1,59 +1,85 @@
-   SET(CMAKE_BUILD_TYPE MinSizeRel)
-ENDIF()
+# Set default build Release
+if(NOT CMAKE_BUILD_TYPE)
+   set(CMAKE_BUILD_TYPE Release)
+endif()
 
-set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -DKXVER=3")
+# Check if necessary configuration is set
+message(STATUS "Generator : ${CMAKE_GENERATOR}")
+message(STATUS "Build Tool : ${CMAKE_BUILD_TOOL}")
+message(STATUS "HDF5 API : $ENV{HDF5_INSTALL_DIR}")
 
-set(MY_LIBRARY_NAME kdbhdf5)
+# Set target library
 file(GLOB C_FILES src/*.c)
 set_source_files_properties(${C_FILES} PROPERTIES LANGUAGE C)
+add_library (${MY_LIBRARY_NAME} SHARED ${C_FILES})
 
-message(STATUS "Generator : ${CMAKE_GENERATOR}")
-message(STATUS "Build Tool : ${CMAKE_BUILD_TOOL}")
-message(STATUS "HDF5 API : $ENV{BUILD_HOME}")
+# Add include directories of this interface and HDF5
+target_include_directories(${MY_LIBRARY_NAME} PRIVATE
+   ${PROJECT_SOURCE_DIR}/include
+   $ENV{HDF5_INSTALL_DIR}/include
+)
 
-include_directories($ENV{BUILD_HOME}/include)
+# Download k.h into include/
+file(DOWNLOAD "https://github.com/KxSystems/kdb/raw/master/c/c/k.h" "${PROJECT_SOURCE_DIR}/include/k.h")
 
+# Search HDF5 library
 find_library(HDF5_LIBRARY
+    REQUIRED
     NAMES hdf5 libhdf5
-    HINTS "$ENV{BUILD_HOME}/lib/" 
+    HINTS "$ENV{HDF5_INSTALL_DIR}/lib/" 
 )
 
+# Set Base CMake C compile flag
+set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -DKXVER=3")
+
+# Link libraries and additional compie flag
 if (MSVC)
     file(DOWNLOAD "https://github.com/KxSystems/kdb/raw/master/w64/q.lib" "${CMAKE_BINARY_DIR}/q.lib" )
-    set(LINK_LIBS "${CMAKE_BINARY_DIR}/q.lib")
+    target_link_libraries(${MY_LIBRARY_NAME} PRIVATE
+        "${CMAKE_BINARY_DIR}/q.lib"
+        ${HDF5_LIBRARY} 
+    )
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W3 /D H5_BUILT_AS_DYNAMIC_LIB /D WIN32_LEAN_AND_MEAN /D _CRT_SECURE_NO_WARNINGS")
 else()
-    set(LINK_LIBS "")
+    target_link_libraries(${MY_LIBRARY_NAME} PRIVATE
+        ${HDF5_LIBRARY}
+    )
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-strict-aliasing")
 endif()
 
-add_library (${MY_LIBRARY_NAME} SHARED ${C_FILES})
-target_link_libraries(${MY_LIBRARY_NAME} ${HDF5_LIBRARY} ${LINK_LIBS})
+# target_link_libraries(${MY_LIBRARY_NAME} ${HDF5_LIBRARY} ${LINK_LIBS})
+
+# Shared library prefix and suffix
+# ex.) telegrafkdb.so for linux
+# Suffix must be `.so` for Mac
+if(APPLE)
+  set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
+endif()
+set_target_properties(${MY_LIBRARY_NAME} PROPERTIES SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
+set_target_properties(${MY_LIBRARY_NAME} PROPERTIES PREFIX "")
 
 # "-undefined dynamic_lookup" needed for building shared libraries on Mac
-# MacOS also produces binaries as .dylib, for kdb these should be ".so"
 # Setting OSFLAG variable needed for installing binary to $QHOME/<QARCH>
 if(APPLE)
    set_target_properties(${MY_LIBRARY_NAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
-   set_target_properties(${MY_LIBRARY_NAME} PROPERTIES SUFFIX .so)
-   set(INSTALL_SCRIPT install.sh)
```

Comment:
Removal of `INSTALL_SCRIPT` set would results in this no longer being included within the releases see the following section of travis build which outlines the items which would be included in the package https://travis-ci.com/github/KxSystems/hdf5/jobs/464127394#L983